v1/plugin/stream_collector_proxy.go

```diff
@@ -62,18 +69,27 @@ func (p *StreamProxy) StreamMetrics(stream rpc.StreamCollector_StreamMetricsServ
 
 	// Error channel where we will forward plugin errors to snap where it
 	// can report/handle them.
-	errChan := make(chan string)
+	p.errChan = make(chan string)
 	// Metrics into the plugin from snap.
-	inChan := make(chan []Metric)
+	p.recvChan = make(chan []Metric)
 	// Metrics out of the plugin into snap.
-	outChan := make(chan []Metric)
+	p.sendChan = make(chan []Metric)
 	// context for communicating that the stream has been closed to the plugin author
 
-	go p.metricSend(outChan, stream)
-	go p.errorSend(errChan, stream)
-	go p.streamRecv(inChan, stream)
+	md, ok := metadata.FromIncomingContext(stream.Context())
+	if !ok {
+		log.Debug("No metadata")
+	}
 
-	return p.plugin.StreamMetrics(stream.Context(), inChan, outChan, errChan)
+	val := []string{"123456789"}
```

Comment:
Let's send `"not-set"` instead of `"123456789"` in the case where the framework isn't setting the task ID.

Also, instead of creating an array of strings consider creating a simple literal to store the task id (i.e. `taskID`).  Inside the conditional below where you check if the map contains the key `"_taskid"` set the literal.  This way you won't have those go routines where you are indexing into an array which could be asking for trouble unless we check the length of the array.

Lastly, consider changing the key `"_taskid"` to `"task-id"`.  I know that this will require a corresponding change in the PR on the framework but unless we can recall why we included the underscore I would remove it.